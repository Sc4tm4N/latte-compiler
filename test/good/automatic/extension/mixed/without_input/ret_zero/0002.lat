
class Node {
  int elem;

  int elem() {
     return elem;
  }

  int elem2() {
    return self.elem;
  }

  int elem3(int[] elem) {
     return elem.length;
  }

  int elem4(int[] elem) {
     return self.elem;
  }
}

int main() {
  Node n1 = new Node;
  printInt(n1.elem());
  printInt(n1.elem2());
  n1.elem = 13;
  printInt(n1.elem3(new int[n1.elem]));
  printInt(n1.elem3(new int[20]));
  printInt(n1.elem4(new int[20]));
  if(n1 == n1)
    return 0 ;
}
