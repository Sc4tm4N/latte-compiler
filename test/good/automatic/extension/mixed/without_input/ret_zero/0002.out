0
0
13
20
13
